src/coreclr/src/binder/assemblybinder.cpp

```diff
@@ -903,11 +924,53 @@ namespace BINDER_SPACE
         }
         else
         {
+            ReleaseHolder<Assembly> pTPAAssembly;
+            SString& simpleName = pRequestedAssemblyName->GetSimpleName();
+
+            // Is assembly in the bundle?
+            // Single-file bundle contents take precedence over TPA.
+            // The list of bundled assemblies is contained in the bundle manifest, and NOT in the TPA.
+            // Therefore the bundle is first probed using the assembly's simple name.
+            // If found, the assembly is loaded from the bundle.  
+            if (Bundle::AppIsBundle())
+            {
+                SString candidates[] = { W(".dll"), W(".ni.dll") };
```

Comment:
Do we still use the `.ni.dll` suffix for anything?

Reply:
@davidwrighton  mentioned [here](https://github.com/dotnet/coreclr/pull/26504#discussion_r321014175) that .ni.dll is used for managed C++ R2R. Has this changed?

Reply:
Well, I think its still probably the way we should make it work. I'm not sure if it works in practice today though. I know it doesn't work in crossgen2.